src/symbolic_dimensions.jl

```diff
@@ -99,35 +99,35 @@ end
 
 
 """
-    uexpand(q::Quantity{<:Any,<:SymbolicDimensions})
+    uexpand(q::AbstractUnionQuantity{<:Any,<:SymbolicDimensions})
 
 Expand the symbolic units in a quantity to their base SI form.
 In other words, this converts a `Quantity` with `SymbolicDimensions`
 to one with `Dimensions`. The opposite of this function is `uconvert`,
 for converting to specific symbolic units, or `convert(Quantity{<:Any,<:SymbolicDimensions}, q)`,
 for assuming SI units as the output symbols.
 """
-function uexpand(q::Q) where {T,R,D<:SymbolicDimensions{R},Q<:AbstractQuantity{T,D}}
+function uexpand(q::Q) where {T,R,D<:SymbolicDimensions{R},Q<:AbstractUnionQuantity{T,D}}
     return convert(constructor_of(Q){T,Dimensions{R}}, q)
 end
 uexpand(q::QuantityArray) = uexpand.(q)
 # TODO: Make the array-based one more efficient
 
 """
-    uconvert(qout::AbstractQuantity{<:Any, <:SymbolicDimensions}, q::AbstractQuantity{<:Any, <:Dimensions})
+    uconvert(qout::AbstractUnionQuantity{<:Any, <:SymbolicDimensions}, q::AbstractUnionQuantity{<:Any, <:Dimensions})
 
 Convert a quantity `q` with base SI units to the symbolic units of `qout`, for `q` and `qout` with compatible units.
 Mathematically, the result has value `q / uexpand(qout)` and units `dimension(qout)`. 
 """
-function uconvert(qout::AbstractQuantity{<:Any, <:SymbolicDimensions}, q::AbstractQuantity{<:Any, <:Dimensions})
+function uconvert(qout::AbstractUnionQuantity{<:Any, <:SymbolicDimensions}, q::AbstractUnionQuantity{<:Any, <:Dimensions})
```

Comment:
Since last review this method (along with other methods for `uconvert`, along with a method for `uexpand`) generalized from just `AbstractQuantity` to `AbstractUnionQuantity`, but it doesn't seem like corresponding unit tests were added. I guess code coverage wouldn't catch this, but would it make sense to add unit tests that test this method in a more general setting (e.g. with `qout` being a `GenericQuantity`?)

Reply:
Okay I looped over a lot of the unittests and now test both `Quantity` and `GenericQuantity`. Also added special tests for `uconvert`.